media/js/firefox/new-ios-redirect-helper.js

```diff
@@ -0,0 +1,37 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// create namespace
+if (typeof window.Mozilla === 'undefined') {
+    window.Mozilla = {};
+}
+
+(function(Mozilla) {
+    'use strict';
+
+    Mozilla.FirefoxNewIosRedirectHelper = {
+        getDomainFromURL: function(url) {
+            var matches = url.match(/^https?\:\/\/([^\/?#]+)(?:[\/?#]|$)/i);
```

Comment:
Clearly not my strongest point, either. :grin:

You're correct - it isn't needed. Was trying to be too fancy by accounting for trailing querystring or hash, but there doesn't seem to be a need to care about anything beyond the domain itself. Removing.